### ISSUE TITLE
How can I use custom StyledComponents with TableVirtuoso?

### ISSUE THREAD

**Issue by AndreaEsposit** [petyosi/react-virtuoso#572]
I have been trying to use some custom table components with the TableVirtuoso component, but I can't seem to get it to work. 

Example code:
```
const StyledTable= styled.table`....`;

const StyledHead= styled.thead`
    background-color: grey;
`;

const COMPONENTS: TableComponents = {
    Table: ({ style, ...props }) => <StyledTable{...props} style={{ ...style, width: 700 }} />, // this one works ok
    TableHead: forwardRef<HTMLTableSectionElement>((props, ref) => <StyledHead{...props} ref={ref} />), // this one does not work
    TableHead: StyledHead, // does not work
    TableHead; ()  => <StyledHead/> // does not work
};
```

I don't manage to assign the tableHead component correctly, as it requireds specifically a component of type: ComponentType<{ context?: unknown; }>....

Does TableVirtuoso support only MUI components?

**Comment by petyosi** [comment]
By "does not work" I assume you get a typescript error. That's [what I get here](https://codesandbox.io/s/gracious-drake-kweic?file=/src/App.tsx) if I remove `as any`.

I am honestly surprised by it. Recently, the custom components are receiving a new, optional `context` prop, which TS considers suspicious enough to consider the types incompatible. `as any` works, obviously.

**Comment by AndreaEsposit** [comment]
Yes, sorry, I should have specified that it was a TS error.`As any` does indeed resolve my problem. But may I ask, what is exactly the optional `context` prop meant to do? I have been going through the docs and I can't find anything more than  "context<unknown>". ( It might be that I haven't found it yet. )
